libpromises/dbm_api.c

```diff
@@ -197,39 +304,19 @@ void CloseAllDBExit()
     {
         if (db_handles[i].filename)
         {
-            /* Wait until all DB users are served, or a threshold is reached */
-            int count = 0;
-            ThreadLock(&db_handles[i].lock);
-            while (db_handles[i].refcount > 0 && count < 1000)
-            {
-                ThreadUnlock(&db_handles[i].lock);
-
-                struct timespec sleeptime = {
-                    .tv_sec = 0,
-                    .tv_nsec = 10000000                         /* 10 ms */
-                };
-                nanosleep(&sleeptime, NULL);
-                count++;
-
-                ThreadLock(&db_handles[i].lock);
-            }
-            /* Keep mutex locked. */
-
-            /* If we exited because of timeout make sure we Log() it. */
-            if (db_handles[i].refcount != 0)
-            {
-                Log(LOG_LEVEL_ERR,
-                    "Database %s refcount is still not zero (%d), forcing CloseDB()!",
-                    db_handles[i].filename, db_handles[i].refcount);
-                DBPrivCloseDB(db_handles[i].priv);
-            }
-            else /* TODO: can we clean this up unconditionally ? */
-            {
-                free(db_handles[i].filename);
-                db_handles[i].filename = NULL;
-            }
+            CloseDBInstance(&db_handles[i]);
         }
     }
+    
+    DynamicDBHandles *db_dynamic_handles_list = db_dynamic_handles;
+    while (db_dynamic_handles_list)
+    {
+        DBHandle *handle = db_dynamic_handles_list->handle;
+        CloseDBInstance(handle);
+        db_dynamic_handles_list = db_dynamic_handles_list->next;
+        free(handle);
+    }
+    db_dynamic_handles = NULL;
```

Comment:
Is ``db_dynamic_handles`` static? IOW will entries there always remain until application exit?